From 481d24c73d6224894721036d6b8a02ce1479e243 Mon Sep 17 00:00:00 2001
From: Jean-Christophe Trotin <jean-christophe.trotin@st.com>
Date: Wed, 5 Oct 2016 11:32:41 +0200
Subject: [PATCH 19/23] v4l2object: force DMABUF_IMPORT for output-io-mode

output-io-mode is forced to GST_V4L2_IO_DMABUF_IMPORT per default if
"memory:DMABuf" feature is set in the sink caps.

Change-Id: Ie1d999ac5bf72cc690f24e3176e3686f53d1f4fc
Reviewed-on: https://gerrit.st.com/58627
Tested-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
Reviewed-by: Jean Christophe TROTIN <jean-christophe.trotin@st.com>
---
 sys/v4l2/gstv4l2object.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index deab718..6f36486 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -2847,6 +2847,15 @@ gst_v4l2_object_setup_pool (GstV4l2Object * v4l2object, GstCaps * caps)
     v4l2object->req_mode = GST_V4L2_IO_DMABUF;
   }
 
+  if (V4L2_TYPE_IS_OUTPUT (v4l2object->type) &&
+      gst_caps_features_contains (gst_caps_get_features (caps, 0),
+          GST_CAPS_FEATURE_MEMORY_DMABUF) &&
+      v4l2object->req_mode != GST_V4L2_IO_DMABUF_IMPORT) {
+    GST_WARNING_OBJECT (v4l2object->element,
+        "Forcing output-io-mode to GST_V4L2_IO_DMABUF_IMPORT by default...");
+    v4l2object->req_mode = GST_V4L2_IO_DMABUF_IMPORT;
+  }
+
   /* find transport */
   mode = v4l2object->req_mode;
 
-- 
2.7.4

